elip-ct-descriptors.mediawiki

```diff
@@ -0,0 +1,151 @@
+
+Because the scriptPubKey is revealed on the blockchain, its role in the tagged hash is merely to ensure that blinding keys are not reused across multiple outputs, unless they are identical (in which case the reuse amounts to ordinary address reuse).
+Privacy and security are provided by the untweaked key ''K'', which is never revealed on-chain.
+
+In some circumstances, it may make sense for ''K'' to be derived deterministically from the descriptor data, rather than being chosen and specified separately.
+This reduces implementation complexity and the need to store additional keys, at the cost of turning the descriptor unconditionally into a view descriptor.
+In other words, users of this variant will be unable to provide external parties the ability to derive addresses without ''also'' providing them the ability to unblind coins.
+
+In a future ELIP we will specify a standard deterministic key generation scheme for this case.
+
+==Backwards Compatibility==
+
+Using the <code>ct(slip77(<64-byte hex>), <DESCRIPTOR>)</code> construction, any wallet should be able to express its existing confidential addresses using this new scheme.
+
+==Test Vectors==
+
+The following CT descriptors should be parseable and generate the given addresses:
+
+* Valid Descriptor 1: <code>ct(02dce16018bbbb8e36de7b394df5b5166e9adb7498be7d881a85a09aeecf76b623,elpkh(03774eec7a3d550d18e9f89414152025b3b0ad6a342b19481f702d843cff06dfc4))#y6sgetu5</code>
+** Blinding key: <code>02dce16018bbbb8e36de7b394df5b5166e9adb7498be7d881a85a09aeecf76b623</code>
+** Confidential address: <code>CTEp9vcs3eU7zQoyrAqeu9LwdcB8QtC2igYSWt7dhaEQvDwVCceLHdLrREAeYrhC5Jz9Wedn3JyxMzpo</code>
+** Unconfidential address: <code>2daq3zWYvigZd3i8VmBnFrZd4DPT9iV94EP</code>
+* Valid Descriptor 2: <code>ct(02dce16018bbbb8e36de7b394df5b5166e9adb7498be7d881a85a09aeecf76b623,elwpkh(03774eec7a3d550d18e9f89414152025b3b0ad6a342b19481f702d843cff06dfc4))#h5e0p6m9</code>
+** Blinding key: <code>02dce16018bbbb8e36de7b394df5b5166e9adb7498be7d881a85a09aeecf76b623</code>
+** Confidential address: <code>el1qq0r6pegudzm0tzpszelc34qjln4fdxawgwmgnza63wwpzdy6jrm0grmqvvk2ce5ksnxcs9ecgtnryt7xg34060uctupg60d02</code>
+** Unconfidential address: <code>ert1qpasxxt9vv6tgfnvgzuuy9e3j9lryg6ha53x9q0</code>
+* Valid Descriptor 3: <code>ct(02dce16018bbbb8e36de7b394df5b5166e9adb7498be7d881a85a09aeecf76b623,elsh(wpkh(03774eec7a3d550d18e9f89414152025b3b0ad6a342b19481f702d843cff06dfc4)))#3kvhe0a8</code>
+** Blinding key: <code>02dce16018bbbb8e36de7b394df5b5166e9adb7498be7d881a85a09aeecf76b623</code>
+** Confidential address: <code>AzpsK7uqP1KVEMfDQvBXYUkpHmFagD3W4vaLe1X7uy8MS6nj41kNYnaexuXgx14PcbNnYAqBdCSWcbga</code>
+** Unconfidential address: <code>XQ7ffnJkhMwj1H8Ma6N1vcU9mqAa96wB9w</code>
+* Valid Descriptor 4: <code>ct(02dce16018bbbb8e36de7b394df5b5166e9adb7498be7d881a85a09aeecf76b623,eltr(03774eec7a3d550d18e9f89414152025b3b0ad6a342b19481f702d843cff06dfc4))#ytq9w7f3</code>
```

Comment:
Can we please have a valid test vector for the `bare` variant specifying a private key?
That should be the most practical use case, i.e. allowing unblinding

Reply:
Sure, though I'd need to implement that :P. Give me a day or two. I think I can do it before https://github.com/ElementsProject/rust-elements/pull/166 (which is blocking the reference implementation) gets in.